src/ray/raylet/node_manager.cc

```diff
@@ -1957,30 +1932,29 @@ void NodeManager::FinishAssignedActorTask(Worker &worker, const Task &task) {
               HandleObjectLocal(entry.first);
             }
             HandleActorStateTransition(actor_id, std::move(actor_registration));
-            PublishActorStateTransition(
-                actor_id, new_actor_data,
-                /*failure_callback=*/
-                [](gcs::AsyncGcsClient *client, const ActorID &id,
-                   const ActorTableData &data) {
-                  // Only one node at a time should succeed at creating the actor.
-                  RAY_LOG(FATAL) << "Failed to update state to ALIVE for actor " << id;
-                });
+            auto actor_notification = std::make_shared<ActorTableData>(new_actor_data);
+            // The actor was created before.
+            RAY_CHECK_OK(gcs_client_->Actors().AsyncUpdate(
+                JobID::Nil(), actor_id, actor_notification, on_gcs_callback));
           },
-          [actor_id](ray::gcs::AsyncGcsClient *client, const UniqueID &checkpoint_id) {
+          [actor_id](ray::gcs::RedisGcsClient *client, const UniqueID &checkpoint_id) {
             RAY_LOG(FATAL) << "Couldn't find checkpoint " << checkpoint_id
                            << " for actor " << actor_id << " in GCS.";
           }));
     } else {
       // The actor did not resume from a checkpoint. Immediately notify the
       // other node managers that the actor has been created.
       HandleActorStateTransition(actor_id, ActorRegistration(new_actor_data));
-      PublishActorStateTransition(
-          actor_id, new_actor_data,
-          /*failure_callback=*/
-          [](gcs::AsyncGcsClient *client, const ActorID &id, const ActorTableData &data) {
-            // Only one node at a time should succeed at creating the actor.
-            RAY_LOG(FATAL) << "Failed to update state to ALIVE for actor " << id;
-          });
+      auto actor_notification = std::make_shared<ActorTableData>(new_actor_data);
```

Comment:
`actor_notification` isn't set. also, better rename to `actor_data`.

Reply:
Just conflicting again.

Reply:
sorry, I read it wrong. didn't see that `actor_notification` is copied from `new_actor_data`. 
But why not just change `new_actor_data` to `shared_ptr` and avoid a copy?
btw, this variable is "new actor table data to update", not actually a notification. `actor_data`'s meaning sounds clear to me, we're using `task_data` elsewhere as well. But if you don't like it, what about `actor_table_data`.

Reply:
There are other functions which use `new_actor_data`, so i didn't change it to `shared_ptr`.
I think we can change `actor_notification` to a suit name when we rename `ActorTableData`. Must be something better than `actor_data`.

Reply:
You mean `ActorRegistration()`? It accepts a reference, so you can still pass the `shared_ptr` as a reference to it. 
Btw, `node_manager.cc` still has conflict. I guess it's because you only merged `ant-tech-alliance/master`, but not `ray-project/master`. I just updated `ant-tech-alliance/master`. Can you try merging it again?